mesheryctl/internal/cli/root/relationships/list.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// represents the mesheryctl exp relationships list command
+var listRelationshipsCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List registered relationships",
+	Long:  "List all relationships registered in Meshery Server",
+	Example: `
+	View list of relationship
+    mesheryctl exp relationship list
+    View list of relationship with specified page number (25 relationships per page)
+    mesheryctl exp relationship list --page 2
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		// Check prerequisites for the command here
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp relationship list \nRun 'mesheryctl exp relationship list --help' to see detailed help message"
+		if len(args) != 0 {
+			return utils.ErrInvalidArgument(errors.New(errMsg))
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return errors.New(utils.RelationshipsError("this command takes no arguments\n", "list"))
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		var url string
+		if cmd.Flags().Changed("page") {
+			url = fmt.Sprintf("%s/api/meshmodels/relationships?page=%d", baseUrl, pageNumberFlag)
+		} else {
+			url = fmt.Sprintf("%s/api/meshmodels/relationships?pagesize=all", baseUrl)
+		}
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		relationshipsResponse := &models.MeshmodelRelationshipsAPIResponse{}
+		err = json.Unmarshal(data, relationshipsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Entity Details", "Type"}
+		rows := [][]string{}
+
+		for _, relationship := range relationshipsResponse.Relationships {
+			if len(relationship.GetEntityDetail()) > 0 {
+				rows = append(rows, []string{relationship.GetEntityDetail(), string(relationship.Type())})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no relationship is found
+			fmt.Println("No relationship(s) found")
+			return nil
+		}
+
+		if cmd.Flags().Changed("page") {
```

Comment:
For me hitting enter and going to next page causes issue .

Reply:
It's still in draft

Reply:
> For me hitting enter causes issue .

Sorry about that, I still in draft because I was waiting for the interactive pagination to be merged and it was yesterday. I'm working on completing it today